### PR TITLE
Add mutagenesis

### PR DESCRIPTION
**Pull request by xwatkins**
### Reference to existing issue
- A new endpoint is now available to retrieve additional mutagenesis data. As data was loaded at the "category" level, this meant we needed to now load and transform data at the track level.
- Empty categories were being displayed

### Description of changes
- Changed the configuration so that _url_ and _adapters_ are at the track level instead of category level
- Added the new _mutagenesis_ track to the configuration
- Separated data fetching from data manipulation
- Categories are hidden by default, and only shown if they have data to display

### Testing/Styleguide
 - [x] Unit tests pass
 - [x] Type tests pass
 - [x] Linters pass

### CODE REVIEW

File: src/protvista-uniprot.ts [hunk 1]
```diff
@@ -209,68 +217,102 @@ class ProtvistaUniprot extends LitElement {
...
+      // Get the data for all urls and store it
+      for (const url of uniqueUrls) {
+        try {
+          const data = await load(url.replace('{accession}', accession));
+          this.rawData[url] = data.payload;
+        } catch (e) {
+          // TODO handle this better based on error code
+          // Fail silently for now
+        }
+      }
```
Comment (aurel-l) [inline src/protvista-uniprot.ts:237]: Not sure about waiting for the previous data before storing it and then fetching the second data, and so on and so on. We could probably fetch everything at once and let user browser parallelise as it sees fit.
I would probably replace with something like this:
```suggestion
      await Promise.all(
        uniqueUrls.map((url) =>
          load(url.replace('{accession}', accession)).then(
            (data) => (this.rawData[url] = data.payload),
            // TODO handle this better based on error code
            // Fail silently for now
            (error) => console.warn(error)
          )
        )
      );
```
Make sure one error is not blocking the function to continue. If it does, the whole block of code probably needs a try/catch

**Review by aurel-l — CHANGES_REQUESTED** [review]
Looks good. Function-wise it's working as expected 👍🏽

Fixed an error I found, and moved a bit of the logic around to fix linting and types, so please do have an other look and check that I haven't broken something unexpectedly.

Also, I've proposed a different approach for data loading, but haven't pushed code related to it as it might need further discussion. See related inline comment. This is the only bit that I think might need a change.

File: src/protvista-uniprot.ts [hunk 2]
```diff
@@ -209,68 +217,102 @@ class ProtvistaUniprot extends LitElement {
...
+
+      // Now iterate over tracks and categories, transforming the data
+      // and assigning it as adequate
+      this.config.categories.map(
+        ({ name: categoryName, tracks, trackType }) => {
+          // const categoryData: any = [];
+          const categoryData = tracks.map(
+            ({ data: dataConfig, name: trackName, filter }) => {
+              const { url, adapter } = dataConfig[0]; // TODO handle array
+              const trackData = this.rawData[url] || [];
```
Comment (xwatkins) [inline src/protvista-uniprot.ts:247]: Added `[]` as otherwise it wasn't working in when there was no antigen data (see the second example in index.html).

Comment (aurel-l) [reply src/protvista-uniprot.ts:247]: I guess then that now you don't need the extra check I've added in 71c6be2 then

**Comment by xwatkins** [comment]
> Looks good. Function-wise it's working as expected 👍🏽
> 
> Fixed an error I found, and moved a bit of the logic around to fix linting and types, so please do have an other look and check that I haven't broken something unexpectedly.
> 
> Also, I've proposed a different approach for data loading, but haven't pushed code related to it as it might need further discussion. See related inline comment. This is the only bit that I think might need a change.

Also made a couple of changes after noticing a bug. I had considered using `Promise.all` but discarded it because of the "fail fast" approach but if there are potential performance improvements thanks to parallelisation happy to give it a go.

File: src/protvista-uniprot.ts (hunk 2 above)
Comment (xwatkins) [reply src/protvista-uniprot.ts:247]: I guess not but maybe safer to leave it in, I guess eventually we can have all the data types defined properly by the different adapters and simplify a lot of checks.

File: src/protvista-uniprot.ts (hunk 1 above)
Comment (xwatkins) [reply src/protvista-uniprot.ts]: Yep, definitely running the requests in parallel 👍 
I've tried it with examples with no data and it seems to be ok without the try/catch

**Review by aurel-l — APPROVED** [review]

**Pull request merged**